agent/xds/config.go

```diff
@@ -27,6 +27,12 @@ type ProxyConfig struct {
 	// Note: This escape hatch is compatible with the discovery chain.
 	PublicListenerJSON string `mapstructure:"envoy_public_listener_json"`
```

Comment:
Couldn't we define the custom tracing json as part of the larger override?

What's the use-case for targeting the tracing configuration only?